src/NeoSharp.Core/Network/Server.cs

```diff
@@ -191,7 +238,7 @@ private void ListenForMessages(IPeer peer, CancellationToken cancellationToken)
                 {
                     var message = await peer.Receive();
 
-                    if (peer.IsReady == message.IsHandshakeMessage()) continue;
+                    if (message == null || peer.IsReady == message.IsHandshakeMessage()) continue;
```

Comment:
Should we throw on null messages?

Reply:
when message is null, the next iteration is IsConnected=false, and exit clean

Reply:
I am not clear why ```Receive``` can retrieve null and why it is not throwing.

Reply:
If is not connected or fail, return NULL

Reply:
What is the meaning of the delay of https://github.com/CityOfZion/neo-sharp/pull/108/files#diff-9c7603c91c0a95b8232445300e5d6616R251 ?

Reply:
Yes, we should wait if we received nothing. This is the purpose of the Delay: to poll in intervals if nothing to do.

Reply:
sockets are in blocking mode so the delay is not neccessary, but i ussualy put a thread.sleep(1), and is enough, i don’t understand why we need iAsyncDelay, is not neccessary, and more complicate more a simple job

Reply:
Basically, it is the same thing except Thread.Sleep for thread scheduler looks like the thread is busy and Task.Delay can communicate back

Reply:
Do you think that is not more easy do this with static class, without inject it?

Reply:
No doubts, it is easier but in this case, you cannot write a reliable test.